gsa/src/web/pages/hosts/row.js

```diff
@@ -112,13 +112,26 @@ const Row = ({
   onToggleDetailsClick,
   ...props
 }) => {
-  const {details = {}} = entity;
-  const os_cpe = isDefined(details.best_os_cpe)
-    ? details.best_os_cpe.value
-    : undefined;
+  const {details = {}, identifiers} = entity;
+
+  let os_cpe;
+
+  const os_index = identifiers.findIndex(
+    identifier => identifier.name === 'OS',
+  );
+
+  if (isDefined(details) && isDefined(details.best_os_cpe)) {
+    os_cpe = details.best_os_cpe.value;
+  } else if (isDefined(identifiers) && isDefined(identifiers[os_index])) {
+    os_cpe = identifiers[os_index].value;
+  } else {
+    os_cpe = undefined;
+  }
+
```

Comment:
Could be simplified
```javascript
  if (isDefined(details) && isDefined(details.best_os_cpe)) {
    os_cpe = details.best_os_cpe.value;
  } else if (isDefined(identifiers)) {
    os_cpe = identifiers.find(
      identifier => identifier.name === 'OS',
    );
  } 
```
find returns undefined if no value matches.

Please ensure when using find or findIndex that the core-js import is used in the module to support IE 11. Either include

```javascript
import 'core-js/features/array/find-index';
```
or
```javascript
import 'core-js/features/array/find';
```

Reply:
Done! i put the array.find in a separate constant and checked if its defined, because we need to access the value of that object and it's not always loaded immediately. I wanted to avoid an error.